lib/gollum-lib/file.rb

```diff
@@ -131,36 +151,15 @@ def get_disk_reference(name, commit)
       else
         @on_disk_path = pathname.to_path
       end
-      true
+      @on_disk = true
     end
 
-    # Find a file in the given Gollum repo.
-    #
-    # name    - The full String path.
-    # version - The String version ID to find.
-    # try_on_disk - If true, try to return just a reference to a file
-    #               that exists on the disk.
-    #
-    # Returns a Gollum::File or nil if the file could not be found. Note
-    # that if you specify try_on_disk=true, you may or may not get a file
-    # for which on_disk? is actually true.
-    def find(name, version, try_on_disk = false)
-      checked = name.downcase
-      map     = @wiki.tree_map_for(version)
-      commit  = version.is_a?(Gollum::Git::Commit) ? version : @wiki.commit_for(version)
-
-      if (result = map.detect { |entry| entry.path.downcase == checked })
-        @path    = name
-        @version = commit
-
-        if try_on_disk && get_disk_reference(name, commit)
-          @on_disk = true
-        else
-          @blob = result.blob(@wiki.repo)
-        end
-
-        self
-      end
+    def construct_path(name)
+      path = self.path.include?('/') ? self.path.sub(/\/[^\/]+$/, '/') : ''
```

Comment:
To better express the intent here, you could put the `sub` call into a private method like so:
```ruby
def get_dirpath_from_filepath(filepath)
  filepath.sub(/\/[^\/]+$/, '/')
end
#...
def construct_path(name)
  path = self.path.include?('/') ? get_dirpath_from_filepath(self.path) : ''
  #...
end
```

And, as a related question, couldn't we just replace the `sub` call altogether with a much more expressive call to `File::dirname`? For example:
```ruby
def get_dirpath_from_filepath(filepath)
  ::File.dirname(filepath) << ::File::SEPARATOR
end
```

Reply:
Actually, I just noticed all this was weirdly complicated. `construct_path` was chopping off the filename from `self.path` in order to... then add the filename to it. I managed to get rid off the helper method entirely, and changed the `url_path` method to:
```ruby
    # Public: The url path required to reach this file within the repo.
    #
    # Returns the String url_path
    def url_path
      # Chop off the page_file_dir and first slash if necessary
      @wiki.page_file_dir ? self.path[@wiki.page_file_dir.length+1..-1] : self.path
    end
```

Reply:
Erm, interesting... Do the tests adequately reflect all the cases (pages at root, pages at subdir level, uploaded files, etc.) so that we can be certain that this works properly across the board?

Reply:
Good question. I outputted the url_paths generated in the tests, and there is a nice variety of them, fortunately. I now added tests for url_path when page_file_dir is set. Everything seems to work fine.